addons/mail/static/src/models/messaging_notification_handler/messaging_notification_handler.js

```diff
@@ -181,11 +181,15 @@ function factory(dependencies) {
          * @param {integer[]} param0.data.channel_ids
          */
         async _handleNotificationChannelMessage(param0) {
-            const channelId = param0.channelId;
-            const convertedData = this.env.models['mail.message'].convertData(param0);
-
-            const oldMessage = this.env.models['mail.message'].find(message =>
-                message.id === convertedData.id
+            const data = Object.assign({ model: 'mail.channel', res_id: param0.channelId }, param0);
+            // get is more efficient than find
+            const existingThread = this.env.models['mail.thread'].get(
+                this.env.models['mail.thread']._createRecordLocalId(data)
+            );
+            const convertedData = this.env.models['mail.message'].convertData(data);
+            // get is more efficient than find
+            const oldMessage = this.env.models['mail.message'].get(
+                this.env.models['mail.message']._createRecordLocalId(data)
```

Comment:
Comment makes only sense in the diff, not just looking at resulting code.
Also this is quite hacky way of using `.get()` and `_createRecordLocalId()`, both of which are not currently designed for that. Please make a proper function to do a O(1) get/find with data that uniquely define record of given model.

Reply:
Ok to create indirection for this such as `findFromIdentifyingData`, but internally it will keep using get and local id, just like insert.